### PR TITLE
fix: makefile temp logic

### PR DESCRIPTION
**Pull request by mateusoliveira43**
## Why the changes were made

Makefile would create 4 temporary directories per call.

Makefile would not clean up temporary directory if command failed.

Makefile would not check if temporary directory is empty.

## How the changes were made

Fixed explained problems.

## How to test the changes made

Run Makefile commands. All created temporary directories should be deleted

### CODE REVIEW

**Comment by mateusoliveira43** [comment]
@mpryc review

File: hack/run_in_temp.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	log.Printf("directory to copy into temporary directory: %v", pwd)
+	copy := exec.Command("cp", "--recursive", pwd, tempDir)
+	err = copy.Run()
+	if err != nil {
+		log.Print(err.Error())
+		return 1
+	}
+
+	// cmd := exec.Command("make", "docker-build", "IMG=ttl.sh/oadp-operator-mateus-123456:1h")
+	cmd := exec.Command("make", "help")
```
Comment (mateusoliveira43) [inline hack/run_in_temp.go:46]: tried to create a go script, but could not call make inside it :exploding_head: 
going to delete this prior to merge

File: Makefile [hunk 2]
```diff
@@ -544,3 +535,19 @@ lint: golangci-lint ## Run Go linters checks against all project's Go files.
 .PHONY: lint-fix
 lint-fix: golangci-lint ## Fix Go linters issues.
 	$(GOLANGCI_LINT) run --fix
+
+define run-in-temp
+@{ \
+set -e ;\
+TEMP=$$(mktemp -d) ;\
+echo $$TEMP ;\
```
Comment (mateusoliveira43) [inline Makefile:543]: remove echo

**Comment by mpryc** [comment]
I *really* think that using go script to run shell commands is not the right implementation:

Shell -> build go -> spawn new go process -> execute `commands` in Shell

That's a bit too much. We should use Shell that invokes `commands` directly or the built-in shell equivalents.

**Comment by mpryc** [comment]
Here is implementation using shell script (not yet integrated within Makefile).

Please comment which way to proceed (go vs shell):

https://github.com/openshift/oadp-operator/commit/7ed10e5d1e9f3eff8a7021e57906e8027390a78e#diff-8de5118a56011d553059dd47da58cea16ecd90598ef82d0d64bcba3ce9cc0423

**Comment by mateusoliveira43** [comment]
closing in favor of other PR

**Pull request closed without merging**